### PR TITLE
Upgrade to winston 3

### PR DESCRIPTION
**Pull request by dkuida**

### CODE REVIEW

**Comment by idohalevi** [comment]
@dkuida Thanks for the PR I will try to review it this week

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      2,
+      "as-needed"
+    ],
+    "no-var": 2,
+    "yoda": 2,
+    "no-console":2,
+    "no-const-assign":2,
+    "prefer-const":2
+  }
+}
```
Comment (idohalevi) [inline .eslintrc]: Looking good :)

File: lib/winston-logzio.js [hunk 2]
```diff
@@ -1,57 +1,45 @@
...
-var _assign = require('lodash.assign');
-var winston = require('winston');
-var stringifySafe = require('json-stringify-safe');
-var logzioNodejs = require('logzio-nodejs');
+const util = require('util');
+const _assign = require('lodash.assign');
+const winston = require('winston');
+const Transport = require('winston-transport');
+const {createLogger, transports, format} = winston;
+const stringifySafe = require('json-stringify-safe');
```
Comment (idohalevi) [inline lib/winston-logzio.js]: Please delete unused

File: lib/winston-logzio.js [hunk 3]
```diff
@@ -1,57 +1,45 @@
...
+        super(options);
+        this.name = options.name || 'LogzioLogger';
 
-    this.level = options.level || 'info';
+        this.level = options.level || 'info';
 
-    this.logzioLogger = logzioNodejs.createLogger(options);
-};
-var Logzio = winston.transports.Logzio;
+        this.logzioLogger = logzioNodejs.createLogger(options);
```
Comment (idohalevi) [inline lib/winston-logzio.js]: Needs to be documented in the readme how to use the appender in this version

File: lib/winston-logzio.js [hunk 4]
```diff
@@ -1,57 +1,45 @@
...
-    }
-    catch(ex) {
-        return stringifySafe(json, null, null, function() { });
-    }
-};
+    log (info, callback) {
+        setImmediate(() => {
+            this.emit('logged', info);
+        });
+        let meta = info[Symbol.for('splat')][0];
```
Comment (idohalevi) [inline lib/winston-logzio.js]: If no error is logged there will be a bug here

File: lib/winston-logzio.js [hunk 5]
```diff
@@ -1,57 +1,45 @@
...
+        if (meta instanceof Error) {
+            meta = {error: meta.stack || meta.toString()};
+        }
+        const message = {
+            meta,
+            level: info.level,
+            message: info.message
```
Comment (idohalevi) [inline lib/winston-logzio.js]: What if the log message is a json object? you are ignoring all fields but `message`. Please add all of the fields you get in info and if there is `message` key in info add it under the key `message`.

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "winston-logzio",
   "description": "A winston transport wrapper for logzio",
-  "version": "1.0.8",
+  "version": "2.0.1",
```
Comment (idohalevi) [inline package.json]: Let's make it 3.0.1 :)

**Comment by idohalevi** [comment]
@dkuida 
Do you need help with this?

**Comment by dkuida** [comment]
@idohalevi  not at all - just need to find time :) busy days. hope to get to it over the weekend

**Comment by idohalevi** [comment]
@dkuida 
I need to push it, I used your PR to publish a new one. I'm closing this and continue with the other PR

**Pull request closed without merging**